### ISSUE TITLE
DAT-Core adapter stopped working (for some datasets)

### ISSUE THREAD

**Issue by elisabettai** [ITISFoundation/osparc-issues#1231]
**Long Story Short**
Can't load data from datcore anymore (on osparc.io)
This was reported by an external user via the feedback form: https://z43.fogbugz.com/f/cases/192158

**Expected behavior**
I would expect to see the files available in my dataset and I would select acute_dataset.h5.

**Steps to reproduce**
Prerequisite: have datcore API keys set up (go to https://app.pennsieve.io to create a pair).
1. On osparc.io go to DATA tab
2. Select datcore
3. Select "Influence of vagus nerve stimulation on cardiac activity in pigs"
4. A loading icon will stay there forever

**Additional context**
In Portainer (datcore-adapter service), I found the error below (for some datasets). For other datasets, the listing of files inside the dataset works.
```
INFO: [2024-01-15 16:28:54,018/MainProcess] [uvicorn.access:send(489)]  -  127.0.0.1:50172 - "GET /v0/live HTTP/1.1" 200
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 7-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cff5a0>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:a20e3f64-bcd2-4022-88e6-a31eb05bd7e6/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 8-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cfd2a0>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:749f26ef-93eb-421b-9fa1-5c7ea3ada0dd/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 9-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cfe030>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:4ff5c2d2-8fb9-4b49-b833-887ee2e831c3/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 10-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cfe570>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:1abdb8ef-2274-457c-ac88-0c089d3c1c6c/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 11-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cfe3b0>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:876badc8-4c69-464d-a471-8e7e2971fd9f/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 12-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cff140>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:3d7b8207-876a-4697-ba9b-8e0381e469a8/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
WARNING: [2024-01-15 16:28:58,267/MainProcess] [simcore_service_datcore_adapter.modules.pennsieve:logged_gather(128)]  -  Error in 15-th concurrent task <coroutine object PennsieveApiClient._get_pck_id_files at 0x7f1fd5cfdc40>: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:f8ca473f-d5a5-4c90-996b-1ad7338390ec/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
INFO: [2024-01-15 16:29:01,268/MainProcess] [uvicorn.access:send(489)]  -  10.0.9.4:60458 - "GET /v0/datasets/N%3Adataset%3Adddbce01-bae0-4e2b-85fd-dafb47182afa/files_legacy HTTP/1.1" 500
ERROR: [2024-01-15 16:29:01,268/MainProcess] [uvicorn.error:run_asgi(424)]  -  Exception in ASGI application
Traceback (most recent call last):
  File "/home/scu/.venv/lib/python3.10/site-packages/anyio/streams/memory.py", line 94, in receive
    return self.receive_nowait()
  File "/home/scu/.venv/lib/python3.10/site-packages/anyio/streams/memory.py", line 89, in receive_nowait
    raise WouldBlock
anyio.WouldBlock
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/base.py", line 78, in call_next
    message = await recv_stream.receive()
  File "/home/scu/.venv/lib/python3.10/site-packages/anyio/streams/memory.py", line 114, in receive
    raise EndOfStream
anyio.EndOfStream
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/scu/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/scu/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/base.py", line 108, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/api/middleware_timing.py", line 10, in add_process_time_header
    response = await call_next(request)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/base.py", line 84, in call_next
    raise app_exc
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/base.py", line 70, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/home/scu/.venv/lib/python3.10/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 169, in __call__
    raise exc
  File "/home/scu/.venv/lib/python3.10/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 167, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/scu/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/scu/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/scu/.venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/scu/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/home/scu/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 163, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/scu/.venv/lib/python3.10/site-packages/servicelib/fastapi/requests_decorators.py", line 96, in wrapper
    return await handler_task
  File "/home/scu/.venv/lib/python3.10/site-packages/aiocache/decorators.py", line 93, in wrapper
    return await self.decorator(f, *args, **kwargs)
  File "/home/scu/.venv/lib/python3.10/site-packages/aiocache/decorators.py", line 108, in decorator
    result = await f(*args, **kwargs)
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/api/routes/datasets.py", line 139, in list_dataset_files_legacy
    file_metas = await pennsieve_client.list_all_dataset_files(
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/modules/pennsieve.py", line 409, in list_all_dataset_files
    await logged_gather(
  File "/home/scu/.venv/lib/python3.10/site-packages/servicelib/utils.py", line 140, in logged_gather
    raise error
  File "/home/scu/.venv/lib/python3.10/site-packages/servicelib/utils.py", line 117, in sem_task
    return await task
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/modules/pennsieve.py", line 233, in _get_pck_id_files
    await self._get_package_files(
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/modules/pennsieve.py", line 218, in _get_package_files
    await self._request(
  File "/home/scu/.venv/lib/python3.10/site-packages/tenacity/_asyncio.py", line 86, in async_wrapped
    return await fn(*args, **kwargs)
  File "/home/scu/.venv/lib/python3.10/site-packages/tenacity/_asyncio.py", line 48, in __call__
    do = self.iter(retry_state=retry_state)
  File "/home/scu/.venv/lib/python3.10/site-packages/tenacity/__init__.py", line 351, in iter
    return fut.result()
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/usr/local/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/home/scu/.venv/lib/python3.10/site-packages/tenacity/_asyncio.py", line 51, in __call__
    result = await fn(*args, **kwargs)
  File "/home/scu/.venv/lib/python3.10/site-packages/simcore_service_datcore_adapter/modules/pennsieve.py", line 148, in _request
    response.raise_for_status()
  File "/home/scu/.venv/lib/python3.10/site-packages/httpx/_models.py", line 749, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '404 ' for url 'https://api.pennsieve.io/packages/N:package:a20e3f64-bcd2-4022-88e6-a31eb05bd7e6/files?limit=1&offset=0'
For more information check: https://httpstatuses.com/404
```

**Comment by elisabettai** [comment]
Hi @Olivier-tl, we have a fix for your issue, but it still needs to reach our production deployment (osparc.io), this will take more than a week at least.

Is this problem blocking you? If yes, I can see what we can do to get it on osparc.io earlier.

**Comment by Olivier-tl** [comment]
Hi @elisabettai, thank you for fixing so quickly! 

This issue is not time critical, I will wait for the fix to reach your production deployment on osparc.io.

**Comment by elisabettai** [comment]
Hi @Olivier-tl, I was checking if the issue with your dataset was resolved, but I bumped into [another issue](https://github.com/ITISFoundation/osparc-issues/issues/1268). I am quite sure you're also affected by this new issue.

I will check with the team what we can do to solve. Apologizes again for the inconvenience and thanks a lot for your patience!
